tests/conftest.py

```diff
@@ -26,11 +26,24 @@ def allowed_audiences() -> List[str]:
 
 
 @pytest.fixture(autouse=True)
-def clear_authz_collector():
+def clear_authz_collector() -> Generator:
     yield
     authz_collector.clean()
 
 
+@pytest.fixture(autouse=True)
+def clear_router_collector() -> Generator:
+    yield
+    Router()._clear()  # pylint: disable=protected-access
+
+
+@pytest.fixture()
+def sample_router() -> Router:
```

Comment:
If this fixture is used only in one test module, it is recommended to put it directly there 😉 

Otherwise, we will have a pretty big `conftest.py` which will be hard to maintain.